packages/echo-base/src/errors/NetworkError.ts

```diff
@@ -30,9 +31,34 @@ export class NetworkError extends BaseError {
     };
 }
 
-export class BadRequestError extends NetworkError {}
-export class BackendError extends NetworkError {}
-export class ForbiddenError extends NetworkError {}
-export class UnauthorizedError extends ForbiddenError {}
-export class NotFoundError extends NetworkError {}
-export class ValidationError extends NetworkError {}
+export class BadRequestError extends NetworkError {
+    constructor(args: NetworkErrorArgs) {
+        super({ ...args, name: args.name ?? 'BadRequestError' });
+    }
+}
+
+export class BackendError extends NetworkError {
+    constructor(args: NetworkErrorArgs) {
+        super({ ...args, name: args.name ?? 'BackendError' });
```

Comment:
It would be nice to test these as well: if they default to the given error name as expected (given no `name` is passed)